addons/sale/models/sale_order.py

```diff
@@ -873,23 +873,37 @@ def action_draft(self):
 
     def action_quotation_send(self):
         """ Opens a wizard to compose an email, with relevant mail template loaded by default """
-        self.ensure_one()
-        self.order_line._validate_analytic_distribution()
+        self.filtered(lambda so: so.state in ('draft', 'sent')).order_line._validate_analytic_distribution()
         lang = self.env.context.get('lang')
-        mail_template = self._find_mail_template()
-        if mail_template and mail_template.lang:
-            lang = mail_template._render_lang(self.ids)[self.id]
+
         ctx = {
             'default_model': 'sale.order',
             'default_res_ids': self.ids,
-            'default_template_id': mail_template.id if mail_template else None,
             'default_composition_mode': 'comment',
-            'mark_so_as_sent': True,
             'default_email_layout_xmlid': 'mail.mail_notification_layout_with_responsible_signature',
             'proforma': self.env.context.get('proforma', False),
-            'force_email': True,
-            'model_description': self.with_context(lang=lang).type_name,
         }
+
+        if len(self) > 1:
+            ctx['default_composition_mode'] = 'mass_mail'
+        else:
+            ctx.update({
+                'force_email': True,
+                'model_description': self.with_context(lang=lang).type_name,
+            })
+            if not self.env.context.get('hide_default_template'):
+                mail_template = self._find_mail_template()
+                if mail_template:
+                    ctx.update({
+                        'default_template_id': mail_template.id,
+                        'mark_so_as_sent': True,
```

Comment:
can be added regardless of whether there is a mail template, like it was before.

Reply:
Hello, when marking SO as sent directly as it was done before in this case https://github.com/odoo/odoo/blob/master/addons/website_sale/controllers/main.py#L743
we did not getting anything in recovery cart.

Reply:
Oh indeed, I thought you added the special context for the recovery action too.
Ok then :ok_hand: